### PR TITLE
[docs] added `versionadded` and `versionchanged` for v0.4.2

### PR DESCRIPTION
**Pull request by ydcjeff**
Part of #1575 

Description: added `versionadded` and `versionchanged` for [v0.4.2](https://github.com/pytorch/ignite/releases/tag/v0.4.2)

[1](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/distributed.html#ignite.distributed.auto.auto_model), [2](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/distributed.html#ignite.distributed.auto.auto_optim), [3](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/distributed.html#ignite.distributed.launcher.Parallel.run), [4](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/distributed.html#ignite.distributed.utils.device), [5](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/distributed.html#ignite.distributed.utils.backend), [6](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/distributed.html#ignite.distributed.utils.model_name), [7](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/distributed.html#ignite.distributed.utils.broadcast),  [8](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/distributed.html#ignite.distributed.utils.sync),  [9](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/distributed.html#ignite.distributed.utils.initialize),  [10](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/handlers.html#ignite.handlers.ModelCheckpoint),  [11](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/handlers.html#ignite.handlers.DiskSaver),  [12](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/metrics.html#ignite.metrics.Metric.attach),  [13](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/metrics.html#ignite.metrics.SSIM), [14](https://deploy-preview-1578--pytorch-ignite-preview.netlify.app/contrib/handlers.html#stores)

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by ydcjeff** [comment]
We need to have some conventions about that. For now, I added every change at the end of the docstring as I find it easier to read and find changelog. But, let me know what you think about that.

Make some css changes – bold and italic to be able to find quickly.

File: ignite/handlers/checkpoint.py [hunk 1]
```diff
@@ -694,6 +697,9 @@ class ModelCheckpoint(Checkpoint):
             there must not be another object in ``to_save`` with key ``checkpointer``.
         **kwargs: Accepted keyword arguments for `torch.save` or `xm.save` in `DiskSaver`.
 
+    .. versionchanged:: 0.4.2
+        Accept ``kwargs`` for torch.save or xm.save
```
Comment (ydcjeff) [inline ignite/handlers/checkpoint.py:701]:
```suggestion
        Accept ``kwargs`` for `torch.save` or `xm.save`
```

**Comment by vfdev-5** [comment]
Thanks a lot for the PR @ydcjeff ! 
Looks very pretty now. I'm wondering about two things however:
- what's the best practice of doing things with such versionadded things
- should we keep multiple notes like that:
```
Changed in version 0.4.2: Added another thing

Changed in version 0.4.1: Added something
```
or keep the latest one ?
- Can we aggregate message into a single one for the same version:
```
Changed in version 0.4.2: 
  - Added Horovod distributed framework. 
  - Added sync_bn argument.
```

What do you think ?

**Comment by Devanshu24** [comment]
Sorry to jump in, I feel this could help the discussion
While researching a bit on how to implement deprecation messages in libraries I found [this](https://github.com/tantale/deprecated/blob/master/deprecated/sphinx.py). It seems relevant to what you are discussing here
Maybe I could add something similar when adding the `@deprecated` decorator

Please ignore if it is not helpful :)

**Comment by vfdev-5** [comment]
@Devanshu24 thanks for joining the discussion and there is no restrictions about that, so no worries. Thanks for the link, let me check that too. The only thing I'm a bit worried is that it is a new dependency that we usually tend to avoid if really necessary...

**Comment by Devanshu24** [comment]
> @Devanshu24 thanks for joining the discussion and there is no restrictions about that, so no worries. Thanks for the link, let me check that too. The only thing I'm a bit worried is that it is a new dependency that we usually tend to avoid if really necessary...

No, I am not suggesting to include the full dependency. It does a lot of error checking and that increases it's size, I could try to write a stripped-down version of it, basically, just something that can do the version management using decorators and also be compatible with sphinx

**Comment by ydcjeff** [comment]
@vfdev-5 I noticed this a few days ago while reading python docs. So, I think we should follow what python ecosystem used to.

> - what's the best practice of doing things with such versionadded things

I noticed python docs adds those at the end of the docstring for some and in the middle (before example) for some. But, I find that is harder to read in ignite docs (like `Checkpoint`) where there are many notes in between. So, I would like to add at the end of docstring as a convention as I find it easier to read and find both on the docs (we describe this convention somewhere in the docs, users will become used to it later) and in the IDE. WDYT?

> - should we keep multiple notes like that:

I think we don't need all of them, may be start from 0.4.0? as the changes will not be reflected to previous version docs.

> ```
> Changed in version 0.4.2: 
>  - Added Horovod distributed framework. 
>  - Added sync_bn argument.
> ```
I haven't tried this one tho, will try.

**Comment by vfdev-5** [comment]
@Devanshu24 sounds good !

> So, I would like to add at the end of docstring as a convention as I find it easier to read and find both on the docs (we describe this convention somewhere in the docs, users will become used to it later) and in the IDE. WDYT?

@ydcjeff I agree !

File: docs/source/contrib/handlers.rst [hunk 2]
```diff
@@ -43,6 +43,17 @@ time_profilers
 .. automodule:: ignite.contrib.handlers.time_profilers
    :members:
 
+stores
```
Comment (vfdev-5) [inline docs/source/contrib/handlers.rst:46]: Thanks for catching that !

**Review by vfdev-5 — APPROVED** [review]
Thanks a lot for the PR @ydcjeff ! 
Looks good to me. Something else do you think we need to add here ?

**Comment by ydcjeff** [comment]
I think there's any left to add.

**Pull request merged**